pages/moodEntry.tsx

```diff
@@ -25,7 +29,7 @@ const MoodEntry = () => {
           {/* Have it link to a confirmation window, which redirects to history after 5 seconds */}
           <div className={styles.submit}>
             <Link href="/">
```

Comment:
Having a link that takes you somewhere with an `onClick` on it could be grounds for a race condition. It might not be, but I'd be suspicious. Changing this from a `Link` wrapping an `a` to a `form` whose `onSubmit` function calls `adToStorage` and then navigates using the `router` that you can import from `next is a little cleaner and leverages the browser's native form handling, which is nice (enter submission for the win!). Changing the `Link` with an `a` tag to a `button` of type `submit` is the way to go to make that happen. Don't forget to get the event in the `form`'s `onSubmit` function and start by calling `preventDefault` on it, or the form will try to route you somewhere before executing the rest of the callback.

Reply:
Something like this:

```typescript
const handleFormSubmission = (e: React.SyntheticEvent) => {
  e.preventDefault()
  addToStorage()
  router.push("/")
}

/ perhaps stuff here, then the render call.

<form onSubmit={handleFormSubmission}>
...
```